src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -1121,96 +1225,90 @@ def _getBoto3BlockDeviceMappings(cls, type_info: InstanceType, rootVolSize: int
-        tags = ec2.get_all_tags({'tag:': {_TAG_KEY_TOIL_CLUSTER_NAME: self.clusterName}})
-        idsToCancel = [tag.id for tag in tags]
-        return [request for request in requests if request.id in idsToCancel]
+        requests: List[SpotInstanceRequestTypeDef] = ec2.describe_spot_instance_requests()["SpotInstanceRequests"]
+        tag_filter: FilterTypeDef = {"Name": "tag:" + _TAG_KEY_TOIL_CLUSTER_NAME, "Values": [self.clusterName]}
+        tags: List[TagDescriptionTypeDef] = ec2.describe_tags(Filters=[tag_filter])["Tags"]
+        # tags = ec2.get_all_tags({'tag:': {_TAG_KEY_TOIL_CLUSTER_NAME: self.clusterName}})
+        idsToCancel = [tag["ResourceId"] for tag in tags]
+        return [request["SpotInstanceRequestId"] for request in requests if request["InstanceId"] in idsToCancel]
 
     def _createSecurityGroups(self) -> List[str]:
         """
         Create security groups for the cluster. Returns a list of their IDs.
         """
+        def group_not_found(e: ClientError) -> bool:
+            retry = (get_error_status(e) == 400 and 'does not exist in default VPC' in get_error_body(e))
+            return retry
 
         # Grab the connection we need to use for this operation.
         # The VPC connection can do anything the EC2 one can do, but also look at subnets.
-        vpc = self.aws.boto2(self._region, 'vpc')
+        boto3_ec2: EC2Client = self.aws.client(region=self._region, service_name="ec2")
 
-        def groupNotFound(e):
-            retry = (e.status == 400 and 'does not exist in default VPC' in e.body)
-            return retry
-        # Security groups need to belong to the same VPC as the leader. If we
-        # put the leader in a particular non-default subnet, it may be in a
-        # particular non-default VPC, which we need to know about.
-        vpcId = None
+        vpc_id = None
         if self._leader_subnet:
-            subnets = vpc.get_all_subnets(subnet_ids=[self._leader_subnet])
+            subnets = boto3_ec2.describe_subnets(SubnetIds=[self._leader_subnet])["Subnets"]
             if len(subnets) > 0:
-                vpcId = subnets[0].vpc_id
-        # security group create/get. ssh + all ports open within the group
+                vpc_id = subnets[0]["VpcId"]
         try:
-            web = vpc.create_security_group(self.clusterName,
-                                            'Toil appliance security group', vpc_id=vpcId)
-        except EC2ResponseError as e:
-            if e.status == 400 and 'already exists' in e.body:
-                pass  # group exists- nothing to do
+            if vpc_id is None:
+                web_response: CreateSecurityGroupResultTypeDef = boto3_ec2.create_security_group(GroupName=self.clusterName, Description='Toil appliance security group')
+            else:
+                web_response = boto3_ec2.create_security_group(GroupName=self.clusterName, Description='Toil appliance security group', VpcId=vpc_id)
```

Comment:
This might be better done as building a dict, conditionally adding a key, and `**`ing it into the boto3 call's kwargs.